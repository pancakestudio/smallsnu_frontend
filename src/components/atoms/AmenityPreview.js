import React from 'react'
import { Card, ListGroup } from 'react-bootstrap'
import './AmenityPreview.css'

export const AmenityPreview = ({kr_name, operating_hours, onClick}) => {
  return(
    <Card className="resPreview">
      <Card.Header
        className="cardHeader"
        as="h5"
        onClick={onClick}
        style={{cursor: "pointer"}}
      >
        {kr_name}
      </Card.Header>
      <ListGroup variant="flush">
        <ListGroup.Item>운영 시간: {operating_hours}</ListGroup.Item>
      </ListGroup>
    </Card>
  )
}
