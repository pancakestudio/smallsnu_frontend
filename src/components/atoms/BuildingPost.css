.buildingPost .content:hover {
  overflow: hidden;
}
