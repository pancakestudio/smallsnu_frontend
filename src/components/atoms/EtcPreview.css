.etcPreview .content:hover {
  overflow: hidden;
}
