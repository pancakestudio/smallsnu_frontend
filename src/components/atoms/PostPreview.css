.postPreview {
  overflow: hidden;
}

.postPreview:hover .title{
  text-decoration: underline;
}

.postPreview .footer {
  margin-top: -20px;
  background-color: rgba(0,0,0,0);
  font-size: 14px;
}

@media(max-width: 576px){
  .postPreview .title {
    font-size: calc(10px + 2vmin);
  }

  .postPreview .footer {
    font-size: calc(10px + 0.7vmin);
  }
}
