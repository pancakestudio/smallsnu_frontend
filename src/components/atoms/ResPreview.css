.resPreview {
  margin: 5px 0px;
}
