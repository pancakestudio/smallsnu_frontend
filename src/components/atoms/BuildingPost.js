import React from 'react'
import { Card } from 'react-bootstrap'
import './BuildingPost.css'

export const BuildingPost = ({title, content}) => {
  return(
    <Card className="buildingPost border-right-0 border-left-0 rounded-0">
      <Card.Body>
        <Card.Title className="title">{title}</Card.Title>
        <Card.Text className="content text-muted text-truncate">{content}</Card.Text>
      </Card.Body>
    </Card>
  )
}
