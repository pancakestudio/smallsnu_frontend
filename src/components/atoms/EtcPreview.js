import React from 'react'
import { Card, ListGroup } from 'react-bootstrap'
import './EtcPreview.css'

export const EtcPreview = () => {
  return(
    <Card className = "etc">
      <ListGroup variant="flush">
        <ListGroup.Item>etc</ListGroup.Item>
      </ListGroup>
    </Card>
  )
}
