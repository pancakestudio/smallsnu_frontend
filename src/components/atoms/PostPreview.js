import React from 'react'
import { Card } from 'react-bootstrap'
import './PostPreview.css'

export const PostPreview = ({title, content, onClick}) => {
  return(
    <Card
      className="postPreview border-right-0 border-left-0 rounded-0"
      onClick={onClick}
      style={{cursor: "pointer"}}
    >
      <Card.Body>
        <Card.Title className="title">{title}</Card.Title>
        <Card.Text className="content text-muted text-truncate">{content}</Card.Text>
      </Card.Body>
    </Card>
  )
}
