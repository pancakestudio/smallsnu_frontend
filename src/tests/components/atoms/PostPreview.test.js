import React from 'react'
import { PostPreview } from '../../../components/atoms/PostPreview'
import { shallow } from 'enzyme'

describe('PostPreview', ()=>{
  let component
  const title = 'title'
  const content = 'content'

  it('renders correctly', ()=>{
    component = shallow(
      <PostPreview
        title={title}
        content={content}
      />
    )
  })

  it('matches snapshot', ()=>{
    expect(component).toMatchSnapshot()
  })

  it('has a correct title and content', ()=>{
    expect(component.find('CardTitle').text()).toBe('title')
    expect(component.find('CardText').text()).toBe('content')
  })
})
