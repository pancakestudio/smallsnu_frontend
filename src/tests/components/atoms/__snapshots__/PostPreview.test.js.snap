// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostPreview matches snapshot 1`] = `ShallowWrapper {}`;
